search/services/metadata.py

```diff
@@ -142,6 +144,66 @@ def retrieve(self, document_id: str) -> DocMeta:
         return data
 
 
+    def bulk_retrieve(self, document_ids: List[str]) -> Dict[str, DocMeta]:
+        """
+        Retrieve metadata for an arXiv paper.
+
+        Parameters
+        ----------
+        document_ids : List[str]
+
+        Returns
+        -------
+        dict
+
+        Raises
+        ------
+        IOError
+        ValueError
+        """
+        if not document_ids:    # This could use further elaboration.
+            raise ValueError('Invalid value for document_ids')
+        
+        query_string = '?' + '&'.join(f'id={document_id}' for document_id in document_ids)
+
+        try:
+            target = urljoin(self.endpoint, query_string)
+            logger.debug(
+                f'{document_ids}: retrieve metadata from {target} with SSL'
+                f' verify {self._verify_cert}'
+            )
+            response = self._session.get(target, verify=self._verify_cert)
+        except requests.exceptions.SSLError as e:
+            logger.error('SSLError: %s', e)
+            raise SecurityException('SSL failed: %s' % e) from e
+        except requests.exceptions.ConnectionError as e:
+            logger.error('ConnectionError: %s', e)
+            raise ConnectionFailed(
+                'Could not connect to metadata service: %s' % e
+            ) from e
+
+        if response.status_code not in \
+                [status.HTTP_200_OK, status.HTTP_206_PARTIAL_CONTENT]:
```

Comment:
Thinking out loud: what would actually happen if we got a 206? Would we risk not being able to deserialize the content?

Reply:
In which case it would be handled below, I suppose

Reply:
Also thinking out loud here: we should consider adding a parameter to cap the number of `document_ids` per request to the bulk_docmeta endpoint, similar to what id being done with bulk indexing.